src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -279,7 +352,22 @@ DatabaseAndTable DatabaseCatalog::getTableImpl(
     if (!table_id)
     {
         if (exception)
-            exception->emplace(ErrorCodes::UNKNOWN_TABLE, "Cannot find table: StorageID is empty");
+        {
+            String exception_message;
+            if (!db_and_table.first)
+            {
+                DatabaseNameHints hints(*this);
```

Comment:
I think it's better to store an object with hints in `DatabaseCatalog` and reuse it.